kahuna/public/js/util/batch-tracking.js

```diff
@@ -58,5 +59,6 @@ export const trackAll = async ($rootScope, key, input, tasks, emit) => {
   $rootScope.$broadcast("events:batch-operations:complete", { key });
 
   $rootScope.$emit(emit, successes);
+  idleTimeout(() => { $rootScope.$apply(); });
```

Comment:
Oh wow, so the recipients of the events are not triggering dispatch at the end of a batch? Interesting, good to see this. Might be worth a comment to indicate why this is necessary?

Reply:
The reason is angular.